homeassistant/components/notify/kodi.py

```diff
@@ -4,76 +4,99 @@
 For more details about this platform, please refer to the documentation at
 https://home-assistant.io/components/notify.kodi/
 """
+import asyncio
 import logging
+
+import aiohttp
 import voluptuous as vol
 
-from homeassistant.const import (ATTR_ICON, CONF_HOST, CONF_PORT,
-                                 CONF_USERNAME, CONF_PASSWORD)
-from homeassistant.components.notify import (ATTR_TITLE, ATTR_TITLE_DEFAULT,
-                                             ATTR_DATA, PLATFORM_SCHEMA,
-                                             BaseNotificationService)
+from homeassistant.const import (
+    ATTR_ICON, CONF_HOST, CONF_PORT, CONF_USERNAME, CONF_PASSWORD, CONF_SSL)
+from homeassistant.components.notify import (
+    ATTR_TITLE, ATTR_TITLE_DEFAULT, ATTR_DATA, PLATFORM_SCHEMA,
+    BaseNotificationService)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
 import homeassistant.helpers.config_validation as cv
 
+REQUIREMENTS = ['jsonrpc-async==0.4']
+
 _LOGGER = logging.getLogger(__name__)
-REQUIREMENTS = ['jsonrpc-requests==0.3']
 
 DEFAULT_PORT = 8080
+DEFAULT_SSL = False
```

Comment:
The meaning of SSL is a bit different here than it is in the cases where it is opt-out instead of opt-in. In other cases we are talking about **verifying** SSL certificates. In this case it is about if we want to connect over https instead of http. This is only possible if the user has set up some proxy in front of it's Kodi instance.

I think that:

- We should default to `False` because it's an advanced feature
- We should rename it to `USE_SSL` and make sure we mention in the docs that it's only needed when a user has set this up.